lib/webhdfs/client_v1.rb

```diff
@@ -359,7 +408,7 @@ def request(host, port, method, path, op=nil, params={}, payload=nil, header=nil
         end
       end
 
-      if @kerberos and res.code == '307'
+      if @kerberos and res.code == '307' and not params.key?('delegation')
```

Comment:
https://github.com/apache/hadoop/blob/trunk/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/token/delegation/web/MultiSchemeDelegationTokenAuthenticationHandler.java#L128-L183

if delegation token param settled, namenode do not response WWW-Authenticate header

Reply:
Could you add a comment about it in the code?